api/client/request.go

```diff
@@ -258,17 +261,30 @@ func (r *Request) Do() *Response {
 		req.Header.Set("Access-Token", r.accesstoken)
 	}
 
-	resp, err = r.client.Do(req)
-	if err != nil {
-		return &Response{err: err}
-	}
-	if resp.Body != nil {
-		defer resp.Body.Close()
-		body, err = ioutil.ReadAll(resp.Body)
-	}
-	if err != nil {
-		return &Response{err: err}
+	for i := 0; ; i++ {
+		resp, err = r.client.Do(req)
```

Comment:
To test this, you can write a temporary http handler in httptest, and have it delay or return the status code needed to start the timer.